hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/ReplicationConfig.java

```diff
@@ -182,7 +182,12 @@ static ReplicationConfig parseWithoutFallback(ReplicationType type,
       try {
         factor = ReplicationFactor.valueOf(Integer.parseInt(replication));
       } catch (NumberFormatException ex) {
-        factor = ReplicationFactor.valueOf(replication);
+        try {
+          factor = ReplicationFactor.valueOf(replication);
+        } catch (IllegalArgumentException e) {
+          throw new IllegalArgumentException(replication +
+              " is not supported for RATIS replication type", e);
```

Comment:
`type` could be `STAND_ALONE`, too.

```suggestion
              " is not supported for " + type + " replication type", e);
```